### PR TITLE
Replace all deprecated occurences of "sle_version_at_least"

### PR DESCRIPTION
**Pull request by okurz**
Used the command

```
git grep -l 'version_at_least' | xargs sed -i -e "s/\(\w\+\)_version_at_least(.\(.[^'\"]\+\)/is_\1('\2+/g"
```

for the initial conversion and manual tweaking afterwards.

The main goal of this change is to get rid of `sle_version_at_least`. It
can not help much with the challenge to handle all the different
versions and distributions that we want to support.

Related progress issue: https://progress.opensuse.org/issues/38453

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -497,7 +497,7 @@ sub install_to_other_at_least {
     set_var("REAL_INSTALLED_VERSION", $real_installed_version);
     bmwqemu::save_vars();
 
-    return sle_version_at_least($version, version_variable => "REAL_INSTALLED_VERSION");
+    return is_sle($version, version_variable => +"REAL_INSTALLED_VERSION");
```
Comment (kravciak) [inline lib/utils.pm]: Extra `+`?

**Comment by okurz** [comment]
Yeah, need to review all changes

**Comment by rwx788** [comment]
@okurz progress ticket you've mentioned is not that much related, but yeah, we have to review each single change, as sle_version_at_least returns true for all non-sle repos, which is the original reason why we haven't replaced all the usages. But that's good first step.

**Comment by okurz** [comment]
@kravciak Yes, I followed your suggestion but not (only) because it should be more readable but also because of the different behaviour of `sle_version_at_least` which also returned true e.g. for Tumbleweed. The updated version should be more explicit to only handle older SLE versions as such.

Reviewed all former occurences of sle_version_at_least manually and updated PR.

**Comment by asdil12** [comment]
@okurz could you rebase?

File: lib/main_common.pm [hunk 2]
```diff
@@ -634,7 +634,7 @@ sub lxdestep_is_applicable {
 
 sub is_smt {
     # Smt is replaced with rmt in SLE 15, see bsc#1061291
-    return ((get_var("PATTERNS", '') || get_var('HDD_1', '')) =~ /smt/) && !sle_version_at_least('15');
+    return ((get_var("PATTERNS", '') || get_var('HDD_1', '')) =~ /smt/) && !is_sle('15+');
```
Comment (okurz) [inline lib/main_common.pm]: is_sle('<15')

File: lib/utils.pm [hunk 3]
```diff
@@ -497,7 +497,7 @@ sub install_to_other_at_least {
     set_var("REAL_INSTALLED_VERSION", $real_installed_version);
     bmwqemu::save_vars();
 
-    return sle_version_at_least($version, version_variable => "REAL_INSTALLED_VERSION");
+    return is_sle($version + '+', version_variable => "REAL_INSTALLED_VERSION");
```
Comment (rwx788) [inline lib/utils.pm]: I suppose it has to be `.` and not `+`

File: lib/utils.pm [hunk 4]
```diff
@@ -1095,7 +1095,7 @@ Since SLE 15 gdm is running on tty2, so we change behaviour for it and
 openSUSE distris, except for Xen PV (bsc#1086243).
 =cut
 sub get_root_console_tty {
-    return (sle_version_at_least('15') && !is_caasp && !check_var('VIRSH_VMM_TYPE', 'linux')) ? 6 : 2;
+    return (is_sle('15+') && !is_caasp && !check_var('VIRSH_VMM_TYPE', 'linux')) ? 6 : 2;
```
Comment (rwx788) [inline lib/utils.pm]: That will break tests, it should be at least `!is_sle('<15')`, as includes openSUSE

File: lib/utils.pm [hunk 5]
```diff
@@ -1154,7 +1154,7 @@ sub ensure_serialdev_permissions {
     # reboot an alternative https://superuser.com/a/609141/327890 would need
     # handling of optional sudo password prompt within the exec
     # Need backwards support for SLES11-SP4 here, the command "gpasswd" and "stat" are only available with SLES-12 at least.
-    if (is_sle && check_var('VERSION', '11-SP4')) {
+    if (is_sle('11-SP4')) {
```
Comment (rwx788) [inline lib/utils.pm]: Looks like this will die with `Unsupported version parameter`, we should put `=`.

File: lib/x11test.pm [hunk 6]
```diff
@@ -598,15 +585,14 @@ sub exit_firefox {
...
-    my $is_sle_12_sp1          = (check_var('DISTRI', 'sle') && check_var('VERSION', '12-SP1'));
     my $workaround_repetitions = 5;
     my $i                      = $workaround_repetitions;
     my $settings_menu_loaded   = 0;
 
     # the loop is a workaround for SP1: bug in launcher. Sometimes it doesn't react to click
     # The bug will be NOT fixed for SP1.
     do {
-        if ($is_sle_12_sp1) {
+        if (is_sle('12-SP1')) {
```
Comment (rwx788) [inline lib/x11test.pm]: same here, need `=`, or change in `check_version` subroutine.

File: lib/x11test.pm [hunk 7]
```diff
@@ -620,9 +606,9 @@ sub start_gnome_settings {
         wait_still_screen(3);
         $settings_menu_loaded = check_screen('settings', 0);
         $i--;
-    } while ($is_sle_12_sp1 && !$settings_menu_loaded && $i > 0);
+    } while (is_sle('12-SP1') && !$settings_menu_loaded && $i > 0);
```
Comment (rwx788) [inline lib/x11test.pm]: Same as above

File: lib/x11test.pm [hunk 8]
```diff
@@ -620,9 +606,9 @@ sub start_gnome_settings {
...
 
-    if (!$is_sle_12_sp1 || $settings_menu_loaded) {
+    if (!is_sle('12-SP1') || $settings_menu_loaded) {
```
Comment (rwx788) [inline lib/x11test.pm]: Same as above

**Pull request closed without merging**